combat-trainer.lic

```diff
@@ -3883,13 +3908,13 @@ class GameState
   # This avoids repeatedly picking a high rank skill whose learn rate
   # will be 0/34 because not being challenged and consequently
   # starving lower rank skills from being chosen.
-  def sort_by_rate_then_rank(skills, priorities = [])
-    skills.sort_by do |skill|
-      [
-        DRSkill.getxp(skill),                 # choose first by skills with lowest learning rate
-        priorities.include?(skill) ? -1 : 0,  # and of those, choose skills that are to be prioritized
-        DRSkill.getrank(skill)                # and lastly, choose by skills with lowest ranks
-      ]
+  def sort_by_rate_then_rank(skills)
+    skills.sort do |skillA, skillB|
+      compare = DRSkill.getxp(skillA) <=> DRSkill.getxp(skillB)
+      if compare == 0
+        compare = DRSkill.getrank(skillA) <=> DRSkill.getrank(skillB)
+      end
+      compare
     end
   end
```

Comment:
⚡ Merge conflict, this is inadvertently reverting changes introduced in https://github.com/rpherbig/dr-scripts/pull/4686 that should be kept